Src/Plugins/Security/SEC101_011.SquareCredentialsValidator.cs

```diff
@@ -43,5 +60,74 @@ protected override IEnumerable<ValidationResult> IsValidStaticHelper(Dictionary<
 
             return new[] { validationResult };
         }
+
+        protected override ValidationState IsValidDynamicHelper(ref Fingerprint fingerprint,
+                                                                ref string message,
+                                                                Dictionary<string, string> options,
+                                                                ref ResultLevelKind resultLevelKind)
+        {
+            string id = fingerprint.Id;
+            string secret = fingerprint.Secret;
+
+            const string codeForRequest = "123";
+            const string uri = "https://connect.squareup.com/oauth2/token";
+            const string codeNotFoundMessage = "Authorization code not found";
+
+            try
+            {
+                HttpClient client = CreateHttpClient();
```

Comment:
>CreateHttpClient [](http://example.com/codeflow?start=36&length=16)

This code doesn't actually create a client, does it? i.e., we implemented a caching scheme underneath the covers?

If so, this is a bad name for an API. It should be CreateOrRetrieveCachedHttpClient or something...

Reply:
let's rename in a new pr. but, yes, we should rename it :)